### PR TITLE
Show track and team revision when listing races

### PR DESCRIPTION
**Pull request by ebadyano**
Closes https://github.com/elastic/rally/issues/665

### CODE REVIEW

File: esrally/metrics.py [hunk 1]
```diff
@@ -1314,11 +1314,13 @@ def from_dict(cls, d):
         else:
             user_tags = {}
 
-        # Don't restore a few properties like cluster because they (a) cannot be reconstructed easily without knowledge of other modules
+        team_revision = d.get("cluster").get("team-revision")
```
Comment (danielmitterdorfer) [inline esrally/metrics.py]: If there is no `cluster` property this will fail because `#get()` will return `None`.

File: esrally/metrics.py [hunk 2]
```diff
@@ -1160,11 +1160,11 @@ def format_dict(d):
     races = []
     for race in race_store(cfg).list():
         races.append([race.trial_id, time.to_iso8601(race.trial_timestamp), race.track, format_dict(race.track_params), race.challenge_name, race.car_name,
-                      format_dict(race.user_tags)])
+                      format_dict(race.user_tags), race.track_revision, race.cluster.get("team-revision")])
```
Comment (danielmitterdorfer) [inline esrally/metrics.py]: fyi: I thought whether we might want to change the order here and group track-related properties and team-related properties but those two properties are so special that it's probably ok to leave them at the end.

File: esrally/metrics.py (hunk 1 above)
Comment (ebadyano) [reply esrally/metrics.py]: thank you for  catching this!

File: esrally/metrics.py [hunk 3]
```diff
@@ -1314,7 +1314,9 @@ def from_dict(cls, d):
         else:
             user_tags = {}
 
-        team_revision = d.get("cluster").get("team-revision")
+        team_revision = None
+        if d.get("cluster"):
```
Comment (danielmitterdorfer) [inline esrally/metrics.py]: I think more idiomatic is `if "cluster" in d:`

**Pull request merged**